File: lib/api/ai_image_api.py

# # Python code example for downloading an image
# # For more details, visit: https://github.com/pollinations/pollinations/blob/master/APIDOCS.md

# import requests;

# def download_image(image_url):
#     # Fetching the image from the URL
#     response = requests.get(image_url)
#     # Writing the content to a file named 'image.jpg'
#     with open('image.jpg', 'wb') as file:
#         file.write(response.content)
#     # Logging completion message
#     print('Download Completed')

# # Image details
# prompt = 'Generate_an_image_of_a_futuristic_rocket_ship_blasting_off' # Prompt for the image
# width = 360
# height = 800
# seed = 702115403 # Each seed generates a new image variation
# model = 'flux' # Using 'flux' as default if model is not provided

# image_url = f"https://pollinations.ai/p/Generate_an_image_of_a_futuristic_rocket_ship_blasting_off?width=360&height=800&seed=702115403&model=flux" 

# download_image(image_url)


# # Using the pollinations pypi package

# ## pip install pollinations

# import pollinations as ai

# model_obj = ai.Model()

# image = model_obj.generate(
#     prompt=f'Generate an image of a futuristic rocket ship blasting off  {ai.realistic}',
#     model=ai.flux,
#     width=360,
#     height=800,
#     seed=702115403
# )
# image.save('image-output.jpg')

# print(image.url)

File: lib/api/ai_image_api.dart
import 'dart:io';
import 'package:http/http.dart' as http;
import 'package:path_provider/path_provider.dart';

Future<String?> downloadImage(String imageUrl) async {
  // Fetching the image from the URL
  final response = await http.get(Uri.parse(imageUrl));

  if (response.statusCode == 200) {
    // Getting the directory to save the image
    final directory = await getApplicationDocumentsDirectory();
    final filePath = '${directory.path}/image.jpg';

    // Writing the content to a file named 'image.jpg'
    final file = File(filePath);
    await file.writeAsBytes(response.bodyBytes);

    // Logging completion message
    return filePath;
  } else {
    return null;
  }
}

String generateImageUrl({
  required String prompt,
  required int width,
  required int height,
  required int seed,
  String model = 'flux',
}) {
  return 'https://pollinations.ai/p/$prompt?width=$width&height=$height&seed=$seed&model=$model';
}
